core/market/decentralized/src/identity.rs

```diff
@@ -0,0 +1,58 @@
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+use ya_client::model::NodeId;
+
+use ya_core_model::identity;
+use ya_service_bus::{typed as bus, RpcEndpoint};
+
+#[derive(thiserror::Error, Debug, Serialize, Deserialize)]
+pub enum IdentityError {
+    #[error("Can't get default identity: {0}.")]
+    GetDefaultIdError(String),
+    #[error("Can't get identity caused by gsb error: {0}.")]
+    GsbError(String),
+    #[error("No default identity!!! It shouldn't happen!!")]
+    NoDefaultId,
+    #[error("Can't list identities. Error: {0}.")]
+    ListError(String),
+}
+
+/// Wraps calls to identity module. It is necessary to mock identity in tests.
+#[async_trait::async_trait(?Send)]
+pub trait IdentityApi: Send + Sync {
+    async fn default_identity(&self) -> Result<NodeId, IdentityError>;
+    async fn list(&self) -> Result<Vec<NodeId>, IdentityError>;
+}
+
+pub struct IdentityGSB;
+
+#[async_trait::async_trait(?Send)]
+impl IdentityApi for IdentityGSB {
+    async fn default_identity(&self) -> Result<NodeId, IdentityError> {
+        Ok(bus::service(identity::BUS_ID)
+            .send(identity::Get::ByDefault)
+            .await
+            .map_err(|e| IdentityError::GsbError(e.to_string()))?
+            .map_err(|e| IdentityError::GetDefaultIdError(e.to_string()))?
+            .ok_or(IdentityError::NoDefaultId)?
+            .node_id)
+    }
+
+    async fn list(&self) -> Result<Vec<NodeId>, IdentityError> {
+        Ok(bus::service(identity::BUS_ID)
+            .send(identity::List {})
+            .await
+            .map_err(|e| IdentityError::GsbError(e.to_string()))?
+            .map_err(|e| IdentityError::ListError(e.to_string()))?
+            .iter()
+            .map(|identity_info| identity_info.node_id)
```

Comment:
what about locked identities? We should probably filter them out. 

I'm not sure if default identity can be locked. Probably yes... In such a case we should cease broadcasting for it probably

Reply:
Are you sure we want to cease broadcasting for locked identities?
That means that Offers from locked identities shouldn't appear on other Nodes? I think it's impossible.

Probably we want to disable negotiations for locked identities (and what more important signing Agreements). But still I don't think, we should filter these identities here

Reply:
I've performed a test using locked identity via yapapi and it has passed. So locked identities are... not locked in sense I'd expect. Let's leave it for now.

I'd expect that locked identity does not allow usage REST APIs at all, ie. no subscriptions, no activities, no debit notes nor invoices. But thats sth which must be discussed on higher level than just Marketplace.

Reply:
Referenced this comment in #529

Reply:
I agree to everything you said, that shouldn't be possible ;)
But I don't think, we shouldn't broadcast Offers, if they were already there.
To discuss later.